packages/react/src/use-check-resource.ts

```diff
@@ -0,0 +1,70 @@
+import type {
+  CheckResourceRequest,
+  CheckResourcesResult,
+  RequestOptions,
+} from "@cerbos/core";
+// eslint-disable-next-line import/no-extraneous-dependencies -- TODO: not sure about this rule, maybe importing from peer deps should be allowed?
+import isEqual from "lodash/isEqual";
+// eslint-disable-next-line import/no-extraneous-dependencies -- TODO: not sure about this rule, maybe importing from peer deps should be allowed?
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { useCerbos } from "./use-cerbos";
+
+export interface UseCheckResourceReturn {
```

Comment:
In fact, with sufficient TypeScript magic, we ought to be able to define a single hook so that the other hooks become one-liners

```ts
export function useCheckResource(
  request: Omit<CheckResourceRequest, "principal">,
  options?: RequestOptions
): AsyncResult<CheckResourcesResult> {
  return useRequest("checkResource", request, options);
}
```

Reply:
I managed to do this with a `@ts-expect-error` in there 🤮. Luckily none of the ugliness is leaking so I think it's fine unless you can perform a better TS-fu here

Reply:
I'll give it a go!

Reply:
So, I wasn't able to eliminate it, I realised it's the same problem I have run into several times elsewhere in the SDK. I did make a couple of tweaks while I was trying to do so - see what you think: https://github.com/hasanayan/cerbos-sdk-javascript/pull/1